### ISSUE TITLE
missing functions

### ISSUE THREAD

**Issue by freevryheid** [freevryheid/duckdb#11]
I've added the c defs for these. I'll add these when I find time.

**Comment by freevryheid** [comment]
still very much a work in progress. development progressing albeit slowly.

**Comment by lucin81** [comment]
I've seen some of the TODOs in the code. I'll try to take on some of these, maybe starting from the chunk functions.

**Comment by freevryheid** [comment]
Didn't realize the size of this when starting out :) Given our time constraints I think we're making good progress.

@lucin81 - I appreciate your assistance with this! To prevent duplication of effort, can we do some planning to delegate tasks for development. If you can prepare a list of tests to address and the functions needed for these I can work on the interface functions This will also help with the layout of the test framework - I think we should break these into separate files as is done on the cpp side. Let me know what you think or have other suggestions.

**Comment by lucin81** [comment]
I agree. It's a nice project, I don't mind if it progresses slowly and it makes sense organising more so we make the most of the time spent on it. 

It's a good idea, the test driven development works well for this and should also help prioritise the most important/most used functionality. I was planning to follow the C API tests which seem quite comprehensive. Making a full list of the functions needed for each test suite (agree on breaking them) is a sizeable task, so the way I propose to go about it is to do that as we move along. One overall list of tests here, then one issue per test with the details of the functions needed to complete that test.

**Comment by lucin81** [comment]
This is the list of test suites and what we have already implemented. Some of them we might not want and some we might want to prioritise, we can discuss here or in individual issues. Let me know if this is what you had in mind. Here there isn't a full list of functions but it should already give a good idea what APIs are needed for which tests. 

- [x] **test_starting_database.cpp**
  - [x] Simple In-Memory DB Start Up and Shutdown
  - [x] Multiple In-Memory DB Start Up and Shutdown
- [ ] **test_capi.cpp** 
  - [ ] Basic test of C API (_Part implemented_)
  - [ ] Test different types of C API
  - [ ] Test errors in C API
  - [ ] Test C API config
  - [ ] Issue #2058: Cleanup after execution of invalid SQL statement causes segmentation fault
  - [ ] Decimal -> Double casting issue
- [ ] **test_capi_data_chunk.cpp**
  - [x] Test table_info incorrect 'is_valid' value for 'dflt_value' column
  - [x] Test Logical Types C API
  - [ ] Test DataChunk C API (_in progress_)
  - [ ] Test DataChunk result fetch in C API
  - [ ] Test DataChunk populate ListVector in C API
- [ ] **capi_table_functions.cpp**
  - [ ] Test Table Functions C API
  - [ ] Test Table Function errors in C API
  - [ ] Test Table Function named parameters in C API
- [ ] **test_capi_appender.cpp**
  - [ ] Test appending into DECIMAL in C API
  - [ ] Test appender statements in C API
  - [ ] Test append timestamp in C API
- [ ] **test_capi_arrow.cpp**
  - [ ] Test arrow in C API
- [ ] **test_capi_complex_types.cpp**
  - [ ] Test decimal types C API
  - [ ] Test enum types C API
  - [ ] Test list types C API
  - [ ] Test struct types C API
- [ ] **test_capi_extract.cpp**
  - [ ] Test extract statements in C API
- [ ] **test_capi_pending.cpp**
  - [ ] Test pending statements in C API
- [ ] **test_capi_prepared.cpp**
  - [ ] Test prepared statements in C API
- [ ] **test_capi_replacement_scan.cpp**
  - [ ] Test replacement scans in C API
  - [ ] Test error replacement scan
- [ ] **test_capi_streaming.cpp**
  - [ ] Test streaming results in C API
  - [ ] Test other methods on streaming results in C API
- [ ] **test_capi_to_decimal.cpp**
  - [ ] Test CAPI duckdb_decimal_as_properties
- [ ] **test_capi_website.cpp**
  - [ ] Test C API examples from the website

**Comment by freevryheid** [comment]
Perfect. Please move this checklist to the readme so we provide a progress report on the front page. We'll flesh this out over time. I think we should also aim to implement some of the helper functions in the node and python api's to provide convenience functions along the lines as suggested by you in issue [#9](https://github.com/freevryheid/duckdb/issues/9).

**Comment by lucin81** [comment]
started creating some of the issues. functions list are auto generated and approximate. but they should capture the large majority of the functions. will probably mix in some types too, so might need some manual cleaning.
